cachi2/interface/cli.py

```diff
@@ -346,6 +360,72 @@ def inject_files(
     )
 
 
+def _prevalidate_sbom_files_args(sbom_files_to_merge: Paths) -> Paths:
+    def enough_files_for_merge(sbom_files_to_merge: Paths) -> Paths:
+        if len(sbom_files_to_merge) < 2:
+            # NOTE: an exception here happens during argument evaluation phase
+            # i.e. outside of handle_errors() decorator. Simply raising
+            # an exception here will not produce correct exit code, thus
+            # the explicit call to exception wrapper.
+            _bail_out_with_error(InvalidInput("Need at least two different SBOM files"))
```

Comment:
Raising `InvalidInput` would have the same effect as wrapping it with `_bail_out_with_error` here, right? (considering the command is annotated with `@handle_errors`). This would avoid the need to extracting out `_bail_out_with_error`.

In case you prefer to keep it extracted, I'd say doing it in a separate commit would be better.

Reply:
No, it won't: with a `raise` exit code is 1, with the wrapper it is set to 2 for `Cachi2Error`s. This exception will happen _before_ arguments are passed to the decorated function, thus outside of `handle_error` scope.